File: web/utils/bool-utils.dart
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

part of NetTango;

class BoolUtils {

  static bool toBool(var b, [bool defaultValue = false]) {
    if (b == null) {
      return defaultValue;
    }
    else if (b is bool) {
      return b;
    }
    else if (b is String) {
      if (b.toLowerCase() == "true" || b.toLowerCase() == "t") {
        return true;
      }
      else if (b.toLowerCase() == "false" || b.toLowerCase() == "f") {
        return false;
      }
    }
    return defaultValue;
  }

  static bool allAreTrue(Iterable<bool> values) {
    return values.reduce( (a, b) => a && b );
  }

}

File: src/utils/bool-utils.ts
// NetTango Copyright (C) Michael S. Horn, Uri Wilensky, and Corey Brady. https://github.com/NetTango/NetTango

class BoolUtils {

  static toBool(b: any, defaultValue: boolean = false): boolean {
    const maybeBool = BoolUtils.toBoolOrNull(b)
    return (b === null) ? defaultValue : b
  }

  static toBoolOrNull(b: any): boolean | null {
    if (b === null) {
      return null
    }
    switch (typeof(b)) {

      case "boolean":
        return b

      case "string":
        const lower = b.toLowerCase()
        if (["t", "true"].includes(lower)) {
          return true
        }
        if (["f", "false"].includes(lower)) {
          return false
        }
        return null

      default:
        return null

    }
  }

  static allAreTrue(values: boolean[]) {
    return values.reduce( (a, b) => a && b )
  }

}

export { BoolUtils }